appcues/src/main/java/com/appcues/ui/component/AppcuesBoxComponent.kt

```diff
@@ -0,0 +1,115 @@
+package com.appcues.ui.component
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material.ProvideTextStyle
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import com.appcues.data.model.ExperiencePrimitive.BoxPrimitive
+import com.appcues.data.model.ExperiencePrimitive.ButtonPrimitive
+import com.appcues.data.model.ExperiencePrimitive.TextPrimitive
+import com.appcues.data.model.styling.ComponentColor
+import com.appcues.data.model.styling.ComponentStyle
+import com.appcues.data.model.styling.ComponentStyle.ComponentHorizontalAlignment
+import com.appcues.data.model.styling.ComponentStyle.ComponentVerticalAlignment
+import com.appcues.ui.LocalAppcuesActionDelegate
+import com.appcues.ui.LocalAppcuesActions
+import com.appcues.ui.extensions.Compose
+import com.appcues.ui.extensions.PrimitiveGestureProperties
+import com.appcues.ui.extensions.getBoxAlignment
+import com.appcues.ui.extensions.getTextStyle
+import com.appcues.ui.extensions.primitiveStyle
+import com.appcues.ui.theme.AppcuesPreview
+import java.util.UUID
+
+@Composable
+internal fun BoxPrimitive.Compose() {
+    Box(
+        modifier = Modifier.primitiveStyle(
+            component = this,
+            gestureProperties = PrimitiveGestureProperties(
+                onAction = LocalAppcuesActionDelegate.current.onAction,
+                actions = LocalAppcuesActions.current,
+                interactionSource = remember { MutableInteractionSource() },
+                indication = rememberRipple(),
+                enabled = remember { true },
+            ),
+            isDark = isSystemInDarkTheme()
+        ),
+        contentAlignment = style.getBoxAlignment(),
+    ) {
+
+        ProvideTextStyle(style.getTextStyle(LocalContext.current, isSystemInDarkTheme())) {
+            items.forEach { it.Compose() }
+        }
+    }
+}
+
+private val items = arrayListOf(
```

Comment:
does this get conditionally compiled in preview only? or can it be set up that way?